omega-config-gui/src/main/java/draylar/omegaconfiggui/api/screen/ScreenBuilder.java

```diff
@@ -0,0 +1,221 @@
+package draylar.omegaconfiggui.api.screen;
+
+import draylar.omegaconfig.api.Config;
+import draylar.omegaconfiggui.impl.ConfigScreen;
+import net.minecraft.client.gui.DrawableHelper;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.text.Text;
+import net.minecraft.util.Identifier;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.*;
+import java.util.function.Supplier;
+import java.util.function.UnaryOperator;
+
+/**
+ * A builder class to create {@link ConfigScreen}s
+ */
+public final class ScreenBuilder {
+    private final Config parentConfig;
+    public final List<ScreenEntry> entries = new ArrayList<>();
+    public final Text title;
+    public final Identifier backgroundTexture;
+    @Nullable
+    private final ScreenBuilder parent;
+    private final Screen modMenuScreen;
+
+    ScreenBuilder(Config parentConfig, Text title, @Nullable Identifier backgroundTexture, @Nullable ScreenBuilder parent, Screen modMenuScreen) {
+        this.parentConfig = parentConfig;
+        this.title = title;
+        this.backgroundTexture = backgroundTexture;
```

Comment:
imo, every optable builder field should not be final/made into the constructor, and instead a setter should be made.
This would diminish the amount of new `create` methods and would reduce greatly the amount of `null` passing.